docs/source/tutorials/day_of_the_week.qmd

```diff
@@ -0,0 +1,309 @@
+    "negbinom_rv",
+    concentration_rv=nb_conc_rv,
+)
+```
+
+4. And finally, we built the model:
+
+```{python}
+# | label: init-model
+hosp_model = model.HospitalAdmissionsModel(
+    latent_infections_rv=latent_inf,
+    latent_hosp_admissions_rv=latent_hosp,
+    I0_rv=I0,
+    gen_int_rv=gen_int,
+    Rt_process_rv=rtproc,
+    hosp_admission_obs_process_rv=obs,
+)
+```
+
+Here is what the model looks like without the day-of-the-week effect:
+
+```{python}
+# | label: fig-output-admissions-padding-and-weekday
+# | fig-cap: Hospital Admissions posterior distribution without weekday effect
+import jax
+import numpy as np
+
+# Model without weekday effect
+hosp_model.run(
+    num_samples=2000,
+    num_warmup=2000,
+    data_observed_hosp_admissions=daily_hosp_admits,
+    rng_key=jax.random.PRNGKey(54),
+    mcmc_args=dict(progress_bar=False),
+)
+
+# Plotting the posterior
+out = hosp_model.plot_posterior(
+    var="latent_hospital_admissions",
+    ylab="Hospital Admissions",
+    obs_signal=np.pad(
+        daily_hosp_admits.astype(float),
+        (gen_int_array.size, 0),
+        constant_values=np.nan,
+    ),
+)
+```
+
+
+## Round 2: Incorporating day-of-the-week effects
+
+We will re-use the infection to admission interval and infection to hospitalization rate from the previous model. But we will also add a day-of-the-week effect. To do this, we will add two additional arguments to the latent hospital admissions random variable: `day_of_the_week_rv` (a `RandomVariable`) and `obs_data_first_day_of_the_week` (an `int` mapping days of the week from 0:6, zero being Monday). The `day_of_the_week_rv`'s sample method should return a vector of length seven; those values are then broadcasted to match the length of the dataset. Moreover, since the observed data may start in a weekday other than Monday, the `obs_data_first_day_of_the_week` argument is used to offset the day-of-the-week effect.
+
+For this example, the effect will be passed as a truncated normal distribution with a mean of 1.0 and a standard deviation of 0.5. The distribution will be truncated between 0.1 and 10.0[^note-other-examples]:
+
+[^note-other-examples]: A similar weekday effect is implemented in its own module, with example code [here](periodic_effects.html).
+
+```{python}
+# | label: weekly-effect
+# Initializing the RV.
+from pyrenew.metaclass import DistributionalRV
+
+dayofweek_effect = DistributionalRV(
+    "dayofweek_effect",
+    dist.TruncatedNormal(loc=1.0, scale=0.5, low=0.1, high=10.0).expand([7]),
+)
+```
+
+Now, by re-defining the latent hospital admissions random variable with the day-of-the-week effect, we can build a model that includes this effect:
+
+```{python}
+# | label: latent-hosp-weekday
+latent_hosp_wday_effect = latent.HospitalAdmissions(
+    infection_to_admission_interval_rv=inf_hosp_int,
+    infect_hosp_rate_rv=hosp_rate,
+    day_of_week_effect_rv=dayofweek_effect,
+    # As the dataset starts on October 30th, 2023 (Monday)
```

Comment:
Would be nice to write this in a generic way that uses the date library to get the day of the week from the start date and compute the correct integer for the offset.

Reply:
Fixed